From 54ea289ca3a1b589b03e85ea84987da6eacd47ff Mon Sep 17 00:00:00 2001
From: lei zhilin <leizhilin@eswincomputing.com>
Date: Wed, 16 Apr 2025 13:26:22 +0800
Subject: [PATCH 625/652] fix(2d):suspend failed,cherry-pick 0284684 from pm

Changelogs:
    1.Problem cause: first platform_device register not record private struct info galDevice;
    2.modify: not use platform_set_drvdata in suspend/resume/shutdown pm control;

Signed-off-by: lei zhilin <leizhilin@eswincomputing.com>
---
 .../os/linux/kernel/gc_hal_kernel_driver.c    | 28 +++++++++----------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_driver.c b/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_driver.c
index 81dc4fcb6a6e..b7481fdb4a01 100644
--- a/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_driver.c
+++ b/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_driver.c
@@ -1521,7 +1521,7 @@ static void viv_dev_shutdown(struct platform_device *pdev)
 {
     gckGALDEVICE gal_device;
 
-    gal_device = platform_get_drvdata(pdev);
+    gal_device = galDevice;
     if (!gal_device) {
         return;
     }
@@ -1538,10 +1538,10 @@ static int viv_dev_suspend(struct platform_device *dev, pm_message_t state)
     gckDEVICE device;
     gctUINT i, dev_index;
 
-    gal_device = platform_get_drvdata(dev);
-
-    if (!gal_device)
+    gal_device = galDevice;
+    if (!gal_device) {
         return -1;
+    }
 
     for (dev_index = 0; dev_index < gal_device->args.devCount; dev_index++) {
         device = gal_device->devices[dev_index];
@@ -1551,14 +1551,14 @@ static int viv_dev_suspend(struct platform_device *dev, pm_message_t state)
                 /* Store states. */
                 status = gckHARDWARE_QueryPowerState(device->kernels[i]->hardware,
                                                      &device->statesStored[i]);
-
-                if (gcmIS_ERROR(status))
+                if (gcmIS_ERROR(status)) {
                     return -1;
+                }
 
                 status = gckHARDWARE_SetPowerState(device->kernels[i]->hardware, gcvPOWER_OFF);
-
-                if (gcmIS_ERROR(status))
+                if (gcmIS_ERROR(status)) {
                     return -1;
+                }
             }
         }
     }
@@ -1574,8 +1574,7 @@ static int viv_dev_resume(struct platform_device *dev)
     gctUINT i, dev_index;
     gceCHIPPOWERSTATE statesStored;
 
-    gal_device = platform_get_drvdata(dev);
-
+    gal_device = galDevice;
     if (!gal_device)
         return -1;
 
@@ -1586,8 +1585,9 @@ static int viv_dev_resume(struct platform_device *dev)
             if (device->kernels[i] != gcvNULL) {
                     status = gckHARDWARE_SetPowerState(device->kernels[i]->hardware, gcvPOWER_ON);
 
-                if (gcmIS_ERROR(status))
+                if (gcmIS_ERROR(status)) {
                     return -1;
+                }
 
                 /* Convert global state to crossponding internal state. */
                 switch (device->statesStored[i]) {
@@ -1609,10 +1609,10 @@ static int viv_dev_resume(struct platform_device *dev)
                 }
 
                 /* Restore states. */
-                    status = gckHARDWARE_SetPowerState(device->kernels[i]->hardware, statesStored);
-
-                if (gcmIS_ERROR(status))
+                status = gckHARDWARE_SetPowerState(device->kernels[i]->hardware, statesStored);
+                if (gcmIS_ERROR(status)) {
                     return -1;
+                }
             }
         }
     }
-- 
2.53.0

